File: CS/DragDropExample/Gallery/CustomGalleryControl.cs
using System;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.Gallery;

namespace DragDropExample.Gallery {
    public class CustomGalleryControl :GalleryControl {
        bool fIsDragDrop;
        internal bool IsDragDrop {
            get { return fIsDragDrop; }
        }

        public void BeginDrag() {
            fIsDragDrop = true;
        }

        public void EndDrag() {
            fIsDragDrop = false;
        }

        protected override void OnMouseLeave(EventArgs e) {
            base.OnMouseLeave(e);
            EndDrag();
        }

        protected override GalleryControlGallery CreateGallery() {
            return new CustomGalleryControlGallery(this);
        }
    }
}

File: VB/DragDropExample/Gallery/CustomGalleryControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.Gallery

Namespace DragDropExample.Gallery
	Public Class CustomGalleryControl
		Inherits GalleryControl
		Private fIsDragDrop As Boolean
		Friend ReadOnly Property IsDragDrop() As Boolean
			Get
				Return fIsDragDrop
			End Get
		End Property

		Public Sub BeginDrag()
			fIsDragDrop = True
		End Sub

		Public Sub EndDrag()
			fIsDragDrop = False
		End Sub

		Protected Overrides Sub OnMouseLeave(ByVal e As EventArgs)
			MyBase.OnMouseLeave(e)
			EndDrag()
		End Sub

		Protected Overrides Function CreateGallery() As GalleryControlGallery
			Return New CustomGalleryControlGallery(Me)
		End Function
	End Class
End Namespace
